dom/nodes/rootNode.html

```diff
@@ -10,6 +10,20 @@
 <body>
 <script>
 
+test(function () {
+    var shadowHost = document.createElement('div');
+    document.body.appendChild(shadowHost);
+
+    var shadowRoot = shadowHost.attachShadow({mode: 'open'});
+    shadowRoot.innerHTML = '<div class="shadowChild">content</div>';
+
+    var shadowChild = shadowRoot.querySelector('.shadowChild');
+    assert_equals(shadowChild.getRootNode({composed: true}), shadowRoot, "getRootNode() must return context object's shadow-including root if options's composed is true");
```

Comment:
This would return document, no?

Reply:
I think you got the argument inversed.

Reply:
@annevk, according to [spec](https://dom.spec.whatwg.org/#dom-node-getrootnode) `The getRootNode(options) method, when invoked, must return context object’s shadow-including root if options’s composed is true, and context object’s root otherwise`, so if `composed: true` should return `shadowRoot`

Reply:
When it's true, it must return the shadow-including root, which can never be a `ShadowRoot` object if you read the definition of that.

Reply:
My mistake, I've addressed it in new commit, PTAL.